shopify_python/google_styleguide.py

```diff
@@ -105,17 +107,27 @@ def __import_full_path_only(self, node):  # type: (astroid.ImportFrom) -> None
 
     def __avoid_global_variables(self, node):  # type: (astroid.Assign) -> None
         """Avoid global variables."""
+
+        def check_assignment(node):
+            if utils.get_global_option(self, 'class-rgx').match(node.name):
+                return  # Type definitions are allowed if the assign to a class name
+
+            if utils.get_global_option(self, 'const-rgx').match(node.name) or \
+               re.match('^__[a-z]+__$', node.name):
+                if isinstance(node.parent.value, astroid.Const):
```

Comment:
Sadly I am not sure this will work. With type checking it would be great to know the type, but I can think of many legitimate cases that I've used in the past that would break this.

```python
_VALID_XYZ_PATTERN = re.compile('^...$')
PAYLOAD_LENGTH = 13
HEADER_LENGTH = 10
MESSAGE_LENGTH = PAYLOAD_LENGTH+HEADER_LENGTH
```

I think this falls in the same category as the class-name check. If the developer uses a naming convention to tell other developers that it's a const, we should believe them.